cmd/admin-handlers.go

```diff
@@ -1415,3 +1416,40 @@ func (a adminAPIHandlers) SetConfigKeysHandler(w http.ResponseWriter, r *http.Re
 	// Send success response
 	writeSuccessResponseHeadersOnly(w)
 }
+
+// TraceHandler - POST /minio/admin/v1/trace
+// ----------
+// The handler sends http trace to the connected HTTP client.
+func (a adminAPIHandlers) TraceHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "HTTPTrace")
+	trcAll := false
+	if a := r.URL.Query().Get("all"); a != "false" {
+		trcAll = true
+	}
+	objectAPI := validateAdminReq(ctx, w, r)
+	if objectAPI == nil {
+		return
+	}
+	traceCh := make(chan trace.Info)
+	doneCh := make(chan struct{})
+	defer close(doneCh)
+	targetID := mustGetUUID()
+
+	globalTrace.Trace(targetID, traceCh, doneCh, trcAll)
+	defer globalTrace.Unsubscribe(targetID)
+	defer globalTrace.UnsubscribePeers(targetID)
```

Comment:
Unsubscribe, UnsubscribePeers are not needed. It can be done in a simpler way (explained in peer-rest-client.go)